scripts/cbw_pg_func.js

```diff
@@ -14,17 +14,145 @@ function addWatch() {
                 el.addEventListener('click', avaIFaceJS.mapJS.cbw_func.triggerLayer, false);
             }
             for(let el of document.getElementsByName('surface')){
-                el.addEventListener('click', avaIFaceJS.mapJS.cbw_func.changeSurface, false);
+                el.addEventListener('click', avaIFaceJS.cbw_func.changeSurface, false);
             }
+            for(let el of document.getElementsByName("surfDetails")){
+                el.hidden = true;
+            }
+            document.getElementById('surf_combined').hidden = false;
         }
     }
 }
 
+
+function parseDateString(input_date){
+    return new Date(
+        parseInt(input_date.slice(0, 4)),
+        parseInt(input_date.slice(4, 6)),
+        parseInt(input_date.slice(6, 8))
+    )
+}
+
+function toDateString(input_date){
+    return input_date.getDate().toString()
+        .padStart(2, "0") + "-" + input_date.getMonth().toString()
+        .padStart(2, "0") + "-" + input_date.getFullYear().toString();
+}
+
 avaIFaceJS.cbw_func = {
     images:[],
     selected_zone:1,
+    surfaceLayers: [
+        {
+            "name": "combined",
+            legend: {
+                en: '<table><tbody><tr><th>Depth</th><th>Colour</th></tr>' +
+                    '<tr><td>0</td><td><div class="surface_legend" style="background-color:#980056"></div></td></tr>' +
```

Comment:
Slight nitpick, but the color choice here might be too similar to the color used for Depth 25 (#980056 vs #74007a) from a distance.

Reply:
This is true, but cannot be changed because:
1. These colour values are industry standard
2. The two colour values will never appear side-by-side due to the nature of the data.